tests/http/resource/test_async_resource_client_fetch.py

```diff
@@ -0,0 +1,126 @@
+import httpx
+import pytest
+import respx
+
+
+@pytest.mark.asyncio
+async def test_fetch_success(async_resource_client):
+    expected_response = httpx.Response(
+        httpx.codes.OK,
+        json={"data": {"id": "RES-123", "name": "Test Resource", "status": "active"}},
+    )
+
+    with respx.mock:
+        mock_route = respx.get("https://api.example.com/api/v1/test-resource/RES-123").mock(
+            return_value=expected_response
+        )
+
+        resource = await async_resource_client.fetch()
+
+    assert resource.to_dict() == {"id": "RES-123", "name": "Test Resource", "status": "active"}
+    assert mock_route.called
+    assert mock_route.call_count == 1
```

Comment:
We should avoid using call_count == 1. Please, use mock_route.assert_called_once() instead. We can also remove the preview line as it's redundant